packages/synapse-react-client/src/components/AclEditor/AclEditor.test.tsx

```diff
@@ -174,4 +189,183 @@ describe('AclEditor', () => {
+    confirmItem(itemRows[1], mockTeamData2.name, 'Exempt Eligible')
+    confirmItem(
+      itemRows[2],
+      `@${mockUserData1.userProfile?.userName}`,
+      'Can Review & Exempt Eligible',
+    )
+
+    // team 1 is editable and deletable
+    within(itemRows[0]).getByRole('combobox')
+    within(itemRows[0]).getByRole('button')
+    // team2 and user1 are deletable but not editable
+    expect(within(itemRows[1]).queryByRole('combobox')).not.toBeInTheDocument()
+    within(itemRows[1]).getByRole('button')
+    expect(within(itemRows[2]).queryByRole('combobox')).not.toBeInTheDocument()
+    within(itemRows[2]).getByRole('button')
+
+    // Users can still be added
+    expect(queryForAddUserCombobox()).toBeInTheDocument()
+  })
+
+  it('respects the value of canRemoveEntry (boolean)', async () => {
+    const { itemRows } = await setUp({
+      canEdit: true,
+      canRemoveEntry: false,
+    })
+
+    // Data should still be present
+    confirmItem(itemRows[0], mockTeamData.name, 'Can Review')
+    confirmItem(itemRows[1], mockTeamData2.name, 'Exempt Eligible')
+    confirmItem(
+      itemRows[2],
+      `@${mockUserData1.userProfile?.userName}`,
+      'Can Review & Exempt Eligible',
+    )
+
+    // all rows are editable but not deletable
+    within(itemRows[0]).getByRole('combobox')
+    expect(within(itemRows[0]).queryByRole('button')).not.toBeInTheDocument()
+    within(itemRows[1]).getByRole('combobox')
+    expect(within(itemRows[1]).queryByRole('button')).not.toBeInTheDocument()
+    within(itemRows[2]).getByRole('combobox')
+    expect(within(itemRows[2]).queryByRole('button')).not.toBeInTheDocument()
+
+    // Users can still be added
+    expect(queryForAddUserCombobox()).toBeInTheDocument()
+  })
+
+  it('respects the value of canRemoveEntry (function)', async () => {
+    function allowRemovingOnlyTeam1(resourceAccess: ResourceAccess) {
+      return resourceAccess.principalId === MOCK_TEAM_ID
+    }
+
+    const { itemRows } = await setUp({
+      canEdit: true,
+      canRemoveEntry: allowRemovingOnlyTeam1,
+    })
+
+    // Data should still be present
+    confirmItem(itemRows[0], mockTeamData.name, 'Can Review')
+    confirmItem(itemRows[1], mockTeamData2.name, 'Exempt Eligible')
+    confirmItem(
+      itemRows[2],
+      `@${mockUserData1.userProfile?.userName}`,
+      'Can Review & Exempt Eligible',
+    )
+
+    // team 1 is editable and deletable
+    within(itemRows[0]).getByRole('combobox')
+    within(itemRows[0]).getByRole('button')
+    // team2 and user1 are editable but not deletable
+    within(itemRows[1]).getByRole('combobox')
+    expect(within(itemRows[1]).queryByRole('button')).not.toBeInTheDocument()
+    within(itemRows[2]).getByRole('combobox')
+    expect(within(itemRows[2]).queryByRole('button')).not.toBeInTheDocument()
+
+    // Users can still be added
+    expect(queryForAddUserCombobox()).toBeInTheDocument()
+  })
+
+  it('allows overriding a displayed permission value', async () => {
```

Comment:
E.g. the parent component can change the text to 'Can download' in the case where the public group has READ ('Can view') and the entity allows anonymous download.